rootston/desktop.c

```diff
@@ -97,20 +126,24 @@ bool view_center(struct roots_view *view) {
 
 	int width, height;
 	wlr_output_effective_resolution(output, &width, &height);
+	width /= output->scale;
+	height /= output->scale;
 
-	double view_x = (double)(width - size.width) / 2 + l_output->x;
-	double view_y = (double)(height - size.height) / 2 + l_output->y;
-
+	double view_x = (double)(width - box.width) / 2 + l_output->x;
+	double view_y = (double)(height - box.height) / 2 + l_output->y;
 	view_set_position(view, view_x, view_y);
 
 	return true;
 }
 
 void view_setup(struct roots_view *view) {
-	view_center(view);
-
 	struct roots_input *input = view->desktop->server->input;
+	view_center(view);
 	set_view_focus(input, view->desktop, view);
+	wlr_seat_keyboard_notify_enter(input->wl_seat, view->wlr_surface);
```

Comment:
This is already done in `set_view_focus`: https://github.com/swaywm/wlroots/blob/975b9dc365d5a7bec531522320a1506323575525/rootston/cursor.c#L209